[Tan Yang Yi] iP

yytan25 opened the pull request:
### Duke Chatbot

> "The goal of Computer Science is to build something that will last at least until we've finished building it." - Anonymous [(source)](https://www.comp.nus.edu.sg/~damithch/pages/SE-quotes.htm?type=funnyQuotes)

Ever found yourself being swarmed by deadlines and not even know where to start? Or is your phone calendar just way too cluttered with tasks all throughout the day?

Don't worry, the Duke Chatbot is here to help! Some features include:

- Creation of 4 different tasks (_Task_, _ToDo_, _Deadline_, _Event_)
- Mark tasks as complete/incomplete
- Find tasks using keywords

And the best part? When you exit the chatbot, your tasks will be **saved** so you can come back to it later! How cool is that?? :zany_face:

The Duke Chatbot is an _indispensable_ tool for every Computer Science Student, and not just for productivity! If you want to get a head start in CS2103/T, you can also start building your very own chatbot, right now! Here's how:

1. Fork the repository [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. Check that all files are in order.
3. Since you're advanced enough to be starting this project on your own, I'm sure you'll find your way ~~and build a chatbot that's even more powerful than mine~~

Some features to be implemented in the future:

- [ ] Support for finding tasks due before a specified date and time
- [ ] Support for deleting/marking multiple tasks simultaneously

Here's a sample of the welcome screen. Feel free to change it however you want!
```
public void printSelfIntroduction() {
    System.out.printf("I'm %s. Nice to meet you.%n", this.name);
    System.out.println("I support the following commands:" + "\n");
    printCommands();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

theyangyitan committed:
Completed Level-0 features

theyangyitan committed:
Completed Level-0 features

theyangyitan committed:
Merge remote-tracking branch 'origin/master'

theyangyitan committed:
Updated Level-0 features

yytan2525 committed:
Merge remote-tracking branch 'origin/master'

yytan2525 committed:
Complete Level-1 features

yytan2525 committed:
Complete Level-2 features

yytan2525 committed:
Add OOP elements to Level-2 features

yytan2525 committed:
Add Level-3 features

yytan2525 committed:
Add Level-4 features

yytan2525 committed:
Add run() method

yytan2525 committed:
Enhance list() method and minor QoL changes

yytan2525 committed:
Add Level-5 features

yytan2525 committed:
Add Level-6 features

yytan2525 committed:
Add Java enums

yytan2525 committed:
Add Level-7 features

yytan2525 committed:
Fix minor bugs

yytan2525 committed:
Fix minor bugs

yytan2525 committed:
Add Level-8 features

yytan2525 committed:
Add Level-8 features

yytan2525 committed:
Merge branch 'branch-Level-8'

yytan2525 committed:
Add more OOP

yytan2525 committed:
Add minor QoL changes

yytan2525 committed:
Add minor QoL changes

yytan2525 committed:
Add package 'duke'

yytan2525 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

yytan2525 committed:
Add JUnit Tests

yytan2525 committed:
Improve parsing of date & time input

yytan2525 committed:
Add additional checks for date and time inputs

yytan2525 committed:
Package project as JAR file

yytan2525 committed:
Add Javadoc comments

yytan2525 committed:
Tweak the code to comply with Java coding standard

yytan2525 committed:
Remove redundant method

yytan2525 committed:
Add Level-9 features

yytan2525 committed:
Edit JUnit test cases

yytan2525 committed:
Merge branch 'branch-A-JavaDoc'

yytan2525 committed:
Edit JUnit test cases

yytan2525 committed:
Edit JUnit test cases

yytan2525 committed:
Add Javadoc for new methods

yytan2525 committed:
Add support for Checkstyle

yytan2525 committed:
Add support for Checkstyle

samuelmui8 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime due;
```
Can consider using a noun for the name of the field

samuelmui8 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+     * Also shows the completion status of each task marked with an "[X]", and the
+     * breakdown of completed/incomplete tasks.
+     */
+    public void list() {
```
can consider using a verb / clearer name that describes what the method does

samuelmui8 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,63 @@
+ */
+public class Event extends Task {
+
+    protected LocalDateTime start;
+    protected LocalDateTime end;
```
can consider using nouns eg. startTime, endTime

samuelmui8 reviewed (commented):
Good that javadocs present for all classes and public methods. Can consider adding comments within code blocks to clarify code. Other than some minor naming issues, coding style is generally well adhered to. LGTM.

wui-hong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+            } else if (ui.getInvalidInputCount() == 8) {
+                System.out.println("If you keep giving me nonsense, I'm leaving!");
+            }
+            String msg = sc.nextLine();
```
Perhaps spelling out the entire term might be more appropriate

wui-hong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,304 @@
+public class Duke {
+
+    private final Storage storage;
+    private final TaskList tasks;
```
Perhaps naming it listsOfTasks would be more appropriate considering its type TaskList

wui-hong commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,150 @@
+ * The {@code Storage} class. Deals with writing to, and reading from, text files.
+ */
+public class Storage {
+    private final TaskList tasks;
```
Same comment as previous, about naming this variable considering that its type is `TaskList`

wui-hong reviewed (commented):
LGTM, only have some minor naming improvements that could be made. 👍

yytan2525 committed:
Add Level-10 features

yytan2525 committed:
Add bug fixes

yytan2525 committed:
Add missing JavaDocs

Due to major changes in code structure in `Level-10`, many new public
methods were introduced, and some methods function very differently
from before.

This necessitates the need for an update in the JavaDocs so that users
and reviewers can better understand the intentions of the methods,
improving their understanding of the flow structure.

As a further step towards improving code structure, it is proposed that
we create a new Command interface and model all Commands as classes to
implement the interface. This adds a further layer of abstraction and
improves adaptability of code in the long run.

yytan2525 committed:
Add new package 'javafx'

yytan2525 committed:
Add new package 'javafx'

yytan2525 committed:
Merge remote-tracking branch 'origin/master'

yytan2525 committed:
Generate new JAR file

yytan2525 committed:
Add assertions at critical points

yytan2525 committed:
Improve code quality

yytan25 committed:
Merge pull request #1 from yytan25/branch-A-Assertions

Add assertions at critical points

yytan2525 committed:
Improve code quality

yytan25 committed:
Merge branch 'master' into branch-A-CodeQuality

yytan25 committed:
Merge pull request #2 from yytan25/branch-A-CodeQuality

Improve code quality

yytan2525 committed:
Fix minor bugs

yytan2525 committed:
Merge branch 'branch-A-Assertions'

yytan2525 committed:
Fix minor bugs

yytan2525 committed:
Fix minor bugs

yytan2525 committed:
Detect scheduling anomalies

yytan25 committed:
Merge pull request #3 from yytan25/branch-B-DetectAnomalies

Detect scheduling anomalies

yytan25 committed:
Update README.md

yytan25 committed:
Update README.md

yytan25 committed:
Update README.md

yytan25 committed:
Update README.md

yytan25 committed:
Update README.md

yytan25 committed:
Update README.md

yytan25 committed:
Update README.md

yytan25 committed:
Update README.md

yytan2525 committed:
Update User Guide

yytan2525 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	docs/README.md

yytan2525 committed:
Add product screenshot

yytan2525 committed:
Change package and main class name

yytan2525 committed:
Change package and main class name

yytan2525 committed:
Change package and main class name

yytan2525 committed:
Change package and main class name

yytan25 committed:
Update README.md

yytan25 committed:
Update README.md
